GitUI/ScriptsEngine/ScriptInfo.cs

```diff
@@ -34,12 +34,35 @@ public ScriptInfo()
         /// </summary>
         public string? Icon { get; set; }
 
+        /// <summary>
+        /// Gets or sets the Icon or associated file path
+        /// </summary>
+        public string? IconPathName { get; set; }
+
         /// <summary>
         /// Gets the associated bitmap.
         /// </summary>
         /// <returns>Bitmap image.</returns>
         public Bitmap? GetIcon()
         {
+            if (File.Exists(IconPathName))
+            {
+                if (IconPathName.EndsWith(".ico", StringComparison.OrdinalIgnoreCase))
+                {
+                    System.Drawing.Icon icon = new(IconPathName);
```

Comment:
🤔 We're leaking the `icon` here, it should be disposed.
ditto below

Reply:
Thank you for noticing! I am disappointed in IDEs for not even hinting at me that `Icon` implements `IDisposable`.